datasync/downloader/downloader.go

```diff
@@ -1578,6 +1624,14 @@ func (d *Downloader) commitFastSyncData(results []*fetchResult, stateSync *state
 func (d *Downloader) commitPivotBlock(result *fetchResult) error {
 	block := types.NewBlockWithHeader(result.Header).WithBody(result.Transactions)
 	logger.Debug("Committing fast sync pivot as new head", "number", block.Number(), "hash", block.Hash())
+	if result.StakingInfo != nil {
```

Comment:
Isn't AddStakingInfoToDB necessary process? If this nil checking logic doesn't exit, AddStakingInfoToDB will return an error for StakingInfo == nil case. But, this if branch miss the error handling